src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSONPlugin.java

```diff
@@ -143,17 +143,15 @@ private boolean writeJSONAnnotations(File outputFile, ImageData<BufferedImage> i
             JsonObject objectToExport = new JsonObject();
             JsonArray dictionariesArray = new JsonArray();
 
-            int count = 0;
             for(PathObject annotation : objects) {
                 PathShape pathShape = (PathShape)annotation.getROI();
                 Area area = PathROIToolsAwt.getArea(pathShape);
                 PolygonROI[][] annotationPolygons = PathROIToolsAwt.splitAreaToPolygons(area);
 
                 for(int i = 0; i<annotationPolygons[1].length; i++) {
-                    count++;
                     JsonObject jsonAnnotation = new JsonObject();
-                    jsonAnnotation.addProperty("uid", Integer.toString(count));
-                    jsonAnnotation.addProperty("name", Integer.toString(count));
+                    jsonAnnotation.addProperty("uid", annotation.getName());
```

Comment:
is name guaranteed to be unique?

Reply:
Good catch.

When I made this change, I was only operating under the impression that users would `export` annotations that they `imported` annotations that originated from our internal app.

By default, `QuPath` does not assign any values to `annotation.name`.

As such, if the user does not explicitly assign a name to each annotation, then the `uid` and `name` attributes get dropped, like so:

```jsonc
{
  "SourceSlide": "CMP-02-01-Bx A-1-001.svs",
  "dictionaries": [{
    // "uid" missing
    // "name" missing
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [25641.16796875, 11245.8974609375],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [19162.681640625, 11757.3564453125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [20969.837890625, 15542.1572265625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [20969.837890625, 15508.0595703125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [25641.16796875, 11245.8974609375],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [1.0, 0.0, 0.0],
      "fillColor": [1.0, 0.0, 0.0, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Tumor"
  }, {
    // "uid" missing
    // "name" missing
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [6785.3623046875, 6472.27587890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6785.3623046875, 6506.373046875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6785.3623046875, 6472.27587890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [1.0, 0.0, 0.0],
      "fillColor": [1.0, 0.0, 0.0, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Stroma"
  }, {
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [13400.23828125, 6165.39990234375],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [7944.67041015625, 7767.97314453125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [20697.05859375, 7529.2919921875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [10945.232421875, 7154.2216796875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [13400.23828125, 6165.39990234375],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [1.0, 0.0, 0.0],
      "fillColor": [1.0, 0.0, 0.0, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Stroma"
  }]
}
```

When I explicitly added names, the `uid` and `name` attributes are saved, as expected:

```jsonc
{
  "SourceSlide": "CMP-02-01-Bx A-1-001.svs",
  "dictionaries": [{
    "uid": "Blargh",
    "name": "Blargh",
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [25641.16796875, 11245.8974609375],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [19162.681640625, 11757.3564453125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [20969.837890625, 15542.1572265625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [20969.837890625, 15508.0595703125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [25641.16796875, 11245.8974609375],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [1.0, 0.0, 0.0],
      "fillColor": [1.0, 0.0, 0.0, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Tumor"
  }, {
    "uid": "Monkeys",
    "name": "Monkeys",
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [6785.3623046875, 6472.27587890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6785.3623046875, 6506.373046875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6785.3623046875, 6472.27587890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [0.5882352941176471, 0.7843137254901961, 0.5882352941176471],
      "fillColor": [0.5882352941176471, 0.7843137254901961, 0.5882352941176471, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Stroma"
  }, {
    "uid": "Monkeys",
    "name": "Monkeys",
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [13400.23828125, 6165.39990234375],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6814.525390625, 6504.86962890625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [7944.67041015625, 7767.97314453125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [20697.05859375, 7529.2919921875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [10945.232421875, 7154.2216796875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [13400.23828125, 6165.39990234375],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [0.5882352941176471, 0.7843137254901961, 0.5882352941176471],
      "fillColor": [0.5882352941176471, 0.7843137254901961, 0.5882352941176471, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Stroma"
  }]
}
```

Reply:
Off the top of my head, I see two ways around this:

* I re-instate a similar `count` variable that gets used and incremented in cases where `annotation.getName() == NULL`

OR

* I try and find a library that can generate similar `uid` values that we're using in our internal app.

Thoughts?

Reply:
Pinging @areisle, just so we don't forget about this PR

Reply:
sorry, missed this. I vote we either drop name/uid completely, or try to generate a uuid. at this point I don't see it being necessary that the uids that were imported are maintained, but maybe that will change in future

Reply:
> I vote we either drop name/uid completely

Thought about this a bit, and I think it would break the current `AnnotationContext` implementation for our app, considering it uses the UID as the key for the `annotationToId` object?

> or try to generate a uuid

Addressed in e5569a4

Reply:
Pinging @areisle

Not a huge rush, just wanted to make sure this didn't fall to the wayside =)

For further transparency, I made some annotations in `QuPath`:

![image](https://user-images.githubusercontent.com/16601729/73675866-1caf1200-4668-11ea-9a89-430478ca7354.png)

And exported the annotations using the latest build (e5569a4):

```json
{
  "SourceSlide": "24496.svs",
  "dictionaries": [{
    "uid": "ff944894-a10b-41da-8ca5-a18dff9cd8e6",
    "name": "ff944894-a10b-41da-8ca5-a18dff9cd8e6",
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [5029.78173828125, 5452.5673828125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [3436.33740234375, 7967.2216796875],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [6797.5087890625, 8365.5830078125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [7419.9482421875, 6473.36767578125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [5029.78173828125, 5452.5673828125],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [1.0, 0.0, 0.0],
      "fillColor": [1.0, 0.0, 0.0, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Tumor"
  }, {
    "uid": "8ccc2679-283f-4fdd-a6a6-6b67bff18a8a",
    "name": "8ccc2679-283f-4fdd-a6a6-6b67bff18a8a",
    "path": {
      "applyMatrix": true,
      "segments": [
        [
          [13594.544921875, 5128.89892578125],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [11229.2763671875, 7768.041015625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [15412.0673828125, 8664.353515625],
          [0.0, 0.0],
          [0.0, 0.0]
        ],
        [
          [13594.544921875, 5128.89892578125],
          [0.0, 0.0],
          [0.0, 0.0]
        ]
      ],
      "closed": true,
      "strokeColor": [1.0, 0.0, 0.0],
      "fillColor": [1.0, 0.0, 0.0, 0.5],
      "strokeScaling": false
    },
    "zoom": 1.0,
    "context": [],
    "dictionary": "imported",
    "label": "Tumor"
  }]
}
```